src/rtutils.c

```diff
@@ -604,6 +604,42 @@ JL_DLLEXPORT jl_value_t *jl_argument_datatype(jl_value_t *argt JL_PROPAGATES_ROO
     return (jl_value_t*)dt;
 }
 
+int is_globfunction(jl_value_t *v, jl_datatype_t *dv, jl_sym_t **globname_out) {
+    jl_sym_t *globname = dv->name->mt != NULL ? dv->name->mt->name : NULL;
+    *globname_out = globname;
+    int globfunc = 0;
+    if (globname && !strchr(jl_symbol_name(globname), '#') &&
+        !strchr(jl_symbol_name(globname), '@') && dv->name->module &&
+        jl_binding_resolved_p(dv->name->module, globname)) {
+        jl_binding_t *b = jl_get_module_binding(dv->name->module, globname);
+        // The `||` makes this function work for both function instances and function types.
+        if (b && b->value && (b->value == v || jl_typeof(b->value) == v)) {
+            globfunc = 1;
+        }
+    }
+    return globfunc;
+}
+
+static size_t jl_static_show_x_sym_escaped(JL_STREAM *out, jl_sym_t *name) JL_NOTSAFEPOINT
+{
+    size_t n = 0;
+
+    char *sn = jl_symbol_name(name);
+    int hidden = 0;
+    if (!jl_is_identifier(sn) || jl_is_operator(sn)) {
```

Comment:
Is the `isoperator` check really needed? For example this gives `typeof(Base.:(var"+"))` instead of `typeof(Base.:(+))`.

Reply:
Oops! thanks Jeff. I really think i meant to make this `if (!(jl_is_identifier(sn) || jl_is_operator(sn)))`. Because already `jl_is_identifier("+")` returns false, so i think i added this specifically to _allow_ operators to be okay, but i screwed up the parens. That's what i get for coding late at night 😅 thanks - this is a great catch!

```
julia> @ccall jl_is_identifier("+"::Cstring)::Bool
false

julia> @ccall(jl_is_identifier("+"::Cstring)::Bool) || @ccall(jl_is_operator("+"::Cstring)::Bool)
true
```

i've amended the commit to fix this, and also added an expected output test for this case. 👍  thanks!

Reply:
This is now fixed. Does this sound right to you? Thanks.